controllers/Backend.php

```diff
@@ -1,104 +1,351 @@
+            $myView->render(array('post' => $posts, 'errors' => $errors, 'admins' => $admins));
+        }
     }
 
     public function login()
     {
-        $model_user = new User();
+        $manager_user = new UserManager();
+    
+         if(filter_has_var(INPUT_POST, 'submit')){
+            if(filter_has_var(INPUT_POST, 'email')){
+                $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email', FILTER_VALIDATE_EMAIL)));
+            }
+            if(filter_has_var(INPUT_POST, 'password')){
+                $password = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password')));
+            }
+            
+            $errors = [];
+
+            if(empty($email) || empty($password)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis!";
+            }else if($manager_user->is_admin($email,$password) == 0){
+                $errors['exist']  = "Cet administrateur n'existe pas";
+            }
+                    
+            if(empty($errors)){
+                $_SESSION['admin'] = $email;
+                header("Location: /dashboard");
+
+            }
+        }
+        else
+            $errors =[];
         
-        $page="backend/login";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        if(isset($_SESSION['admin'])){
+            header("location: /dashboard");
+        }
+
+        $myView = new View('login', 'backend');
+        $myView->render(array('errors' => $errors));
     }
     
     public function new()
     {
-        $model_user = new User();
+        $manager_user = new UserManager();
         
-        $page="backend/new";
-        $topbar="backend/topbar";
-        require "views/layout.php";
-    }
+         if(filter_has_var(INPUT_POST, 'submit')){
+            $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email')));
+            $token = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'token')));
+
+            $errors = [];
 
-    public function errorback()
+            if(empty($email) || empty($token)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }else if($manager_user->is_modo($email,$token) == 0){
+                $errors['exist'] = "Ce modérateur n'existe pas";
+            }                
+
+            if(empty($errors)){
+                $_SESSION['admin'] = $email;
+                header("Location:/modification-du-mot-de-passe");
+            }
+            }else
+                $errors =[];
+            
+        $myView = new View('new', 'backend');
+        $myView->render(array('errors' => $errors));
+    }
+    
+    public function logout()
     {
-        $model_user = new User();
-        
-        $page="backend/errorback";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        UNSET($_SESSION['admin']);
+        header("Location: /articles-a-la-une");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function header() is discouraged](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)